### PR TITLE
Added endpoint for product and category

### PR DESCRIPTION
**Pull request by marianlungu0001**
@cristianlucaci pls review

### CODE REVIEW

File: troccproducts/views.py [hunk 1]
```diff
@@ -1,99 +1,176 @@
...
+class ProductViewSet(RestDefaultsMixin,
+					 RetrieveModelMixin,
+					 UpdateModelMixin,
+					 DestroyModelMixin,
+					 CreateModelMixin,
+					 GenericCustomView):
+	"""API endpoint for listing and creating products."""
+	permission_classes = {
+		'GET': (permissions.AllowAny,),
+		'POST': (permissions.AllowAny,),
```
Comment (cristianlucaci) [inline troccproducts/views.py]: Should ANY be allowed to post products, or just auth users?
POST is also available in ProductsViewSet, shoulndt be just one POST endpoint?

File: troccproducts/views.py [hunk 2]
```diff
@@ -1,99 +1,176 @@
...
+		return self.retrieve(request, *args, **kwargs)
+
+	def put(self, request, *args, **kwargs):
+		return self.update(request, *args, **kwargs)
+
+	def post(self, request, *args, **kwargs):
+		return self.create(request, *args, **kwargs)
+
+	def delete(self, request, *args, **kwargs):
+		return self.delete(request, *args, **kwargs)
```
Comment (cristianlucaci) [inline troccproducts/views.py]: Should call self.destroy as it enters recursive call in delete()

File: troccproducts/views.py [hunk 3]
```diff
@@ -1,99 +1,176 @@
...
```
Comment (cristianlucaci) [inline troccproducts/views.py]: Should call self.destroy as it enters recursive call in delete()

**Comment by marianlungu0001** [comment]
@cristianlucaci Updated PR

**Pull request merged**